make/osd.mk

```diff
@@ -0,0 +1,18 @@
+DIR := $(dir $(realpath $(firstword $(MAKEFILE_LIST))))
+INTEGREATLY_OPERATOR_TEST_HARNESS_IMAGE ?= $(REG)/$(ORG)/integreatly-operator-test-harness:latest
+
+.PHONY: image/functional/build
+image/functional/build:
+	go mod vendor
+	docker build . -f Dockerfile.functional -t $(INTEGREATLY_OPERATOR_TEST_HARNESS_IMAGE)
+
+.PHONY: image/functional/push
+image/functional/push:
+	docker push $(INTEGREATLY_OPERATOR_TEST_HARNESS_IMAGE)
+
+.PHONY: image/functional/build/push
+image/functional/build/push: image/functional/build image/functional/push
+
+.PHONY: test/compile/functional
+test/compile/functional:
+	CGO_ENABLED=0 go test -v -c -o integreatly-operator-test-harness.test ./test/functional
```

Comment:
Just to note that we should never really have to use these outside of creating dev images that people want to test themselves.

The following Jiras should cover the ability for our CI to build and push all images created out of this repo:

* https://issues.redhat.com/browse/INTLY-5332
* https://issues.redhat.com/browse/INTLY-5335 

All master artifacts will be synced automatically in that scenario, and also can be setup for any release branches we might be creating in the future.

Reply:
Might also make sense to call the make file something other than osd since you renamed all the targets "functional"